### PR TITLE
backend: add config endpoint

### PR DESCRIPTION
**Pull request by roobre**
This PR adds a new service/endpoint called config, which responds to `GET` requests to `/api/config`.

This endpoint returns a JSON-encoded object that contains key/value pairs that can be used by the frontend to fetch configuration paramenters in runtime.

The key/value pairs that are returned are read from environment variables matching the `QUICKPIZZA_CONF_` prefix. So, for example, the following environment variables:

```
QUICKPIZZA_CONF_FOO=bar
QUICKPIZZA_CONF_SOMETHING_ELSE=very long text
```

Are returned like this:

```json
{
  "foo": "bar",
  "something_else": "very long text"
}
```

### CODE REVIEW

**Comment by theSuess** [comment]
Wouldn't it make more sense to have this be part of the frontend instead of yet another service? Just thinking out loud, I have no real objections to more services.

File: cmd/main.go [hunk 1]
```diff
@@ -75,6 +76,12 @@ func main() {
 		server = server.WithFrontend()
 	}
 
+	if envServe("QUICKPIZZA_CONFIG") {
```
[the remark this answers is not in the document]
Comment (roobre) [reply cmd/main.go:79]: All services are on by default :)
Check out the comment at https://github.com/grafana/quickpizza/blob/6aa0025ce18d3739319da270cc574dd203b22466/cmd/main.go#L68-L73

**Comment by roobre** [comment]
@theSuess Currently the frontend is served as static files from the backend, so it's hard to make it configurable. If you have ideas to work around this (that hopefully do not involve making the frontend a different process or container), they're welcome!

**Comment by theSuess** [comment]
Ah I meant adding it to the same router group as the frontend https://github.com/grafana/quickpizza/blob/main/pkg/http/http.go#L176 since it is intended for use with the frontend. But again, I have no strong feelings one way or another.

**Review by theSuess — APPROVED** [review]

File: cmd/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgzlopes) [reply cmd/main.go:79]: Fair enough!

**Pull request merged**